packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol

```diff
@@ -6,123 +6,116 @@ import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
 /**
  * @title L2OutputOracle
  * @notice The L2 state is committed to in this contract
- * The payable keyword is used on appendL2Output to save gas on the msg.value check.
- * This contract should be deployed behind an upgradable proxy
+ *         The payable keyword is used on appendL2Output to save gas on the msg.value check.
+ *         This contract should be deployed behind an upgradable proxy
  */
 // slither-disable-next-line locked-ether
 contract L2OutputOracle is Ownable {
-    /**********
-     * Events *
-     **********/
+    /**
+     * @notice OutputProposal represents a commitment to the L2 state.
+     *         The timestamp is the L1 timestamp that the output root is posted.
+     *         This timestamp is used to verify that the finalization period
+     *         has passed since the output root was submitted.
+     */
+    struct OutputProposal {
+        bytes32 outputRoot;
+        uint256 timestamp;
+    }
 
-    /// @notice Emitted when an output is appended.
-    event l2OutputAppended(
+    /**
+     * @notice Emitted when an output is appended.
```

Comment:
Event parameters should be documented here, also we're moving away from underscores for names of event params but I can fix that in a different PR.

Reply:
I'll just fix it here.

Reply:
Hmmm, scratch that. I tried it and got a non-obvious go test failure.